proxy/Dockerfile

```diff
@@ -2,25 +2,110 @@
 #
 # Builds a slim runtime image with the conduit-proxy binary.
 
-## Build the rust proxy into a binary.
+ARG RUST_IMAGE=rust:1.23.0
+
+ARG RUNTIME_IMAGE=gcr.io/runconduit/base:2017-10-30.01
+
+## Fetch external dependencies.
+#
+# Mocks out all local code and fetch external dependencies to ensure that external sources
+# are cached.
+FROM $RUST_IMAGE as fetch
+WORKDIR /usr/src/conduit
+COPY proxy/.empty.rs proxy/src/lib.rs
+COPY proxy/.empty.rs proxy/convert/src/lib.rs
+COPY proxy/.empty.rs proxy/futures-mpsc-lossy/src/lib.rs
+COPY proxy/.empty.rs proxy/grpc/src/lib.rs
+COPY proxy/.empty.rs proxy/tower-router/src/lib.rs
```

Comment:
.empty.rs is confusing and distracting. Does `touch` not work here? If not, let's document why we're doing it this way and also include a note about it inside the .empty.rs file itself.